src/samplerbank.cpp

```diff
@@ -29,14 +29,24 @@ void SamplerBank::slotSaveSamplerBank(double v) {
     if (v == 0.0 || m_pPlayerManager == NULL) {
         return;
     }
-
+    QString filefilter = tr("Mixxx Sampler Banks (*.xml)");
     QString samplerBankPath = QFileDialog::getSaveFileName(
             NULL, tr("Save Sampler Bank"),
             QString(),
-            tr("Mixxx Sampler Banks (*.xml)"));
-    if (samplerBankPath.isEmpty()) {
+            tr("Mixxx Sampler Banks (*.xml)"),
+            &filefilter);
+    if (samplerBankPath.isNull() || samplerBankPath.isEmpty()) {
         return;
     }
+    // Manually add extension due to bug in QFileDialog
+    // via https://bugreports.qt-project.org/browse/QTBUG-27186
+    // Can be removed after switch to Qt5
+    QFileInfo fileName(samplerBankPath);
+    if (fileName.suffix().isEmpty()) {
+    	QString ext = filefilter.section(".",1,1);
+    	ext.chop(1);
+        samplerBankPath.append(".").append(ext);
+        }
```

Comment:
**extrem nitpicking mode**

line 45 and 46 contain tabs and the closing `}` is not intened correctly. Could you change that as well.

I'm not sure which editor you use. But assuming vim and you have defined tabs as 4 space characters you can use 'retab' to remove the tabs. To fix the indentation in an automated way use mark all lines of the if statement and hit `=`.